### ISSUE TITLE
Your method corrupts text-generation alignment

### ISSUE THREAD

**Issue by Oguzhanercan** [mit-han-lab/nunchaku#48]
Hi, I really love your solution, but in my experiments, I see that text-generation alignment is really bad. One of your other methods, SANA, works really well but SVDQuant, for some reason we ignore, lost its text-generation alignment capacity. Here are a few examples about it:

Prompt: A 39-year-old male of Asian ethnicity with dark skin. shoulder-length red braided hair. hazel eyes that are round, thin eyebrows that are rounded, a wide nose nose, pouty lips. round face, a slightly wide jawline, full cheeks, a rounded chin. They have a faint scar.

Generation with SVDQuant: 

![example4](https://github.com/user-attachments/assets/23de5eb4-6fe5-4b11-ae5e-cb5789c0be8f)

Generation with SANA: 
![image](https://github.com/user-attachments/assets/909df182-d209-4be0-8b4a-c8309c9cd003)

16 bit BFL Space Generation:

![image](https://github.com/user-attachments/assets/4f31c1c1-9386-4c12-9be1-c9828f023c43)

Prompt: A 31-year-old female of Caucasian ethnicity with olive skin. short black kinky hair. wearing sunglasses, , a wide nose nose, full lips. round face, a rounded jawline, balanced cheeks, a rounded chin. They have a small mole.

Generation with SVDQuant: 

![example1](https://github.com/user-attachments/assets/f4a0470c-f535-44bb-803f-97ba6beca945)

Generation with SANA:
![image](https://github.com/user-attachments/assets/7741de10-3568-498a-a2f9-6c6df0cfe0ba)

16 bit BFL Space Generation:

![image](https://github.com/user-attachments/assets/bbf67d31-f6fb-47ee-8c37-3527a824ba11)

I have test 10 prompts with this strategy. I can say that, the alignment between text and generation for 9 examples out of 10 is not good, BFL space and SANA performs much better. Do you have an idea about fix this? Or what could be the reason? Because nf4 quantization performs better in this case. 

Problem for first test case: SVDQuant generates a woman
Problem for second test case: woman's hair is not kinky

**Comment by lmxyy** [comment]
Thanks for trying our method and the feedback. Could you elaborate more on experiment setting? Specifically, which model are you using (dev or schnell)? How many sampling steps are you using? How large is the classifier-free guidance? Do you use quantized T5?

**Comment by lmxyy** [comment]
I tried your prompt in our [online FLUX.1-schnell demo](https://svdquant.mit.edu/) demo. The result looks good:
<img width="1173" alt="image" src="https://github.com/user-attachments/assets/d70246fd-3aa7-4938-a9d1-22b4b3845c45">

**Comment by Oguzhanercan** [comment]
I am running the code from example.py file. I am using dev model. T5 is not quantized, 28 steps, the resolution is 512x512, the cfg is 3.5. Anyway, I am trying to do more experiments with that but when I run the code for forexample 10 samples, and code finishes (without problem), and when I run the code again, your code allocates 6.3 gb vram at loading weights stage, but then does nothing, just waits, not using gpu (but as I said allocates 6.3gb vram), not using cpu. I cannot terminate the code, sudo kill -9 pid is not even working. Do you have any idea for that, because I cannot do more experiments for you.

System
Ubuntu 22.04, Python3.10, 3090

**Comment by lmxyy** [comment]
Did you mean you ran `example.py` twice? The first run finished successfully, but the second run got stuck? 
Currently, our engine only supports inference with a batch size of 1. Could you try feeding your prompts one at a time?

**Comment by Oguzhanercan** [comment]
I ran the example.py program twice sequentially (not in parallel). The first run worked fine, but the second run got stuck while loading the model. I tested this on two different computers. Could there be a bug in your CUDA-level implementation?

I also noticed a recent commit regarding model loading, so I reinstalled everything with the updated version, but the same problem persists. Could this issue be related to my setup, or is it a known problem?"

**Comment by lmxyy** [comment]
We were not aware of the issue before. Did this mean that your `example.py` can only run once?

**Comment by Oguzhanercan** [comment]
> We were not aware of the issue before. Did this mean that your `example.py` can only run once?

Yes.

**Comment by lmxyy** [comment]
Could you confirm that after the first run, the process was terminated correctly?

**Comment by Oguzhanercan** [comment]
To validate termination, I run the code again, the code completed successfully and there were no error or warnings. Of the completion, I check nvidia-smi to validate gpu memory released, and it was also okay. Is there anything else that I need to check?

**Comment by Oguzhanercan** [comment]
Did you able to reproduce the problem? @lmxyy

**Comment by lmxyy** [comment]
> Did you able to reproduce the problem? [@lmxyy](https://github.com/lmxyy)

Nope. Could you install the latest nunchaku with the wheels to test if the issue still persists? If it does, please contact me at muyangli@mit.edu. We can schedule a joint meeting for debugging.
